chia/util/struct_stream.py

```diff
@@ -11,9 +11,11 @@ class StructStream(int):
     """
 
     def __new__(cls: Any, value: int):
-        bits = struct.calcsize(cls.PACK) * 8
         value = int(value)
-        if value.bit_length() > bits:
+        try:
```

Comment:
We need to be careful because this update will disconnect old peers (1.1.4), when they submit a transaction with negative value additions

Reply:
You might consider doing a `struct.unpack` to do a round-trip and fail if the value comes out different, since we got burned by the `calcsize` check once already. It's slower, sure, but it's C code so it won't be **horribly** slow.

Reply:
```python
    v1 = struct.unpack(cls.PACK, struct.pack(cls.PACK, value)
    if v1 != value:
        raise ValueError(...)
```